tests/benchdnn/ip/ip.cpp

```diff
@@ -278,7 +278,16 @@ void skip_invalid_prb(const prb_t *prb, res_t *res) {}
 
 void setup_cmp(compare::compare_t &cmp, const prb_t *prb, data_kind_t kind,
         const args_t &ref_args) {
-    cmp.set_threshold(0.f);
+    // The nvidia implementation has precision issues in some cases
+    // for large problems with post-op sum
+    if (is_nvidia_gpu()
+            && prb->attr.post_ops.find(attr_t::post_ops_t::kind_t::SUM) != -1
+            && prb->dst_dt() == dnnl_f16) {
+        const float trh = epsilon_dt(prb->dt[2]);
+        cmp.set_threshold(trh);
+    } else {
+        cmp.set_threshold(0.f);
+    }
```

Comment:
Do you know why this difference ? Is sum post-op applied over f32 intermediate value or over f16 values for NV backend?

Reply:
I'd say this change can fly in only in case sum post-op is done through a native cuDNN fusion (single call) with f16 accumulation internally, otherwise, the issue is likely inside the implementation that doesn't convert the output to f32 and accumulate pieces in f32.

Reply:
@sgeor255, thanks for looking into implementation details, that's a good start.
I suggest to conduct a little experiment - make `cublasGemmEx` sum-less and return f32 and append sum post-op (with data copied to a dedicated buffer) with f32 as well (through the `cudnnAddTensor` call) to simulate the proper behavior.
If this experiment goes fine, it would prove that there's something wrong indeed with cuda libraries, and I'll re-iterate on that check again. Probably the same one should exist (or appear) for matmul as well.

Reply:
When changing data addresses the issue it always means rounding/accumulation mechanics stands on its way. Smaller ranges usually lead to situations when final numbers remain exact and conversion to f16/f32 and back don't change the number and the check passes.

When exp number if x.5, in the reality, it can be x.5002, which would be rounded towards `x + 1`, while the library might have `x.4998` which would be rounded towards `x`. That library result can be a product of different effects (accumulation in lower data types, inexact division/multiplication, intermediate roundings, etc.) which can't be examined with cuda.

Reply:
@dzarukin thanks for the suggestion, I tested doing the sum post-op separately with `cudnnAddTensor` and the test is passing for all sum scale values I tested.

Reply:
@sgeor255 thanks for checking it!

Then it likely means non-zero `beta` is not aligned with proper f32 conversions.
I would suggest to keep a "split" implementation version as a gold one and use `beta`-fusion if `acc-mode` is specified to be `relaxed`. `acc-mode` isn't used much inside the library yet, but it can become an essential requirement for cuda/cuDNN/cuBLAS functionalities given observed behavior.